src/custom-element.js

```diff
@@ -489,6 +489,46 @@ function createBaseCustomElementClass(win) {
       ));
     }
 
+    /**
+     * @return {!Promise}
+     * @final
+     */
+    whenLoaded() {
+      return this.signals_.whenSignal(CommonSignals.LOAD_END);
+    }
+
+    /**
+     * Ensure that element is eagerly loaded.
+     *
+     * @param {number=} opt_parentPriority
+     * @return {!Promise}
+     * @final
+     */
+    ensureLoaded(opt_parentPriority) {
+      return this.whenBuilt().then(() => {
+        const resource = this.getResource_();
+        if (this.signals_.get(CommonSignals.LOAD_END)) {
+          return this.whenLoaded();
+        }
+        if (
+          resource.getState() != ResourceState.LAYOUT_SCHEDULED ||
+          resource.isMeasureRequested()
+        ) {
+          resource.measure();
+        }
+        if (!resource.isDisplayed()) {
+          return;
+        }
```

Comment:
Shouldn't this be handled by Resources specifically? It seems like we're entangling CE and Resources again.

Reply:
It's still going in that direction. But I'm doing it so that in the next pull request I can add `if (V2) {}` code here, which will be completely free of `Resources` interference. This class already calls into `Resources` in a few places and they are all easy to find and redo for `V2`. So I'd rather keep these dependencies concentrated in one file then spread them over owners, etc.